src/Autocomplete.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from './types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [noSuggestions, setNoSuggestions] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (inputValue.trim() === '') {
+        setSuggestions(people);
+        setNoSuggestions(false);
+
+        return;
+      }
+
+      const filteredPeople = people.filter(person =>
+        person.name.toLowerCase().includes(inputValue.toLowerCase()),
+      );
+
+      setSuggestions(filteredPeople);
+      setNoSuggestions(filteredPeople.length === 0);
+    }, delay);
+
+    return () => clearTimeout(timeoutId);
+  }, [inputValue, people, delay]);
+
+  const handleSelect = (person: Person) => {
+    setInputValue(person.name);
+    setSelectedPerson(person);
+    setSuggestions([]);
+    onSelected(person);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    if (selectedPerson) {
+      setSelectedPerson(null);
+      onSelected(null);
+    }
+  };
+
+  return (
+    <div className="dropdown is-active">
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={inputValue}
+          onChange={handleInputChange}
+          data-cy="search-input"
+        />
+      </div>
+
+      {suggestions.length > 0 && (
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          <div className="dropdown-content">
+            {suggestions.map(person => (
+              <div
+                key={person.slug}
```

Comment:
The checklist mentions not to generate keys on render. In this case, you are using 'person.slug' as a key which is fine as long as 'slug' is unique for each person and doesn't change over time.